Maestro/src/Microsoft.DotNet.Maestro/Services/VsoParameterGenerator.cs

```diff
@@ -2,31 +2,29 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-using System;
 using System.Collections.Generic;
-using System.Linq;
 using Newtonsoft.Json.Linq;
 
 namespace Microsoft.DotNet.Maestro.Services
 {
     public static class VsoParameterGenerator
     {
-        public static string GetParameters(IDictionary<string, JToken> vsoParameters, Func<string, bool> filter)
+        public static string GetParameters(IDictionary<string, JToken> actionArguments)
         {
-            if (vsoParameters == null || !vsoParameters.Any())
+            JToken vsoBuildParametersToken = null;
+            actionArguments?.TryGetValue("vsoBuildParameters", out vsoBuildParametersToken);
+
+            if (vsoBuildParametersToken == null)
             {
                 return null;
             }
 
-            JObject parameterObject = new JObject();
-            foreach (KeyValuePair<string, JToken> parameter in vsoParameters)
+            JObject parameterObject = (JObject)vsoBuildParametersToken;
```

Comment:
Is it worth throwing a more descriptive error if this isn't a `JObject`? I don't think it'll be a common problem though.